Musl does not have stack unwinder like glibc therefore
we can not assume that its always available on musl, we
do need to check for target environment as well which
could be musl or glibc.

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/mozglue/misc/StackWalk.cpp
+++ b/mozglue/misc/StackWalk.cpp
@@ -44,7 +44,7 @@ using namespace mozilla;
 #  define MOZ_STACKWALK_SUPPORTS_MACOSX 0
 #endif
 
-#if (defined(linux) &&                                            \
+#if (defined(linux) && defined(__GLIBC__) &&                      \
      ((defined(__GNUC__) && (defined(__i386) || defined(PPC))) || \
       defined(HAVE__UNWIND_BACKTRACE)))
 #  define MOZ_STACKWALK_SUPPORTS_LINUX 1
